src/transformStateWithClones.js

```diff
@@ -7,7 +7,43 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const workState = { ...state };
+
+  for (const i of actions) {
+    for (const key in i) {
```

Comment:
Use correct naming. Second loop is redudant. Use switch/case for `action.type`
```suggestion
  for (const action of actions) {
```